src/Easy.zig

```diff
@@ -111,6 +111,67 @@ pub const Response = struct {
                 .name = name,
             };
     }
+
+    pub const HeaderIterator = struct {
+        handle: *c.CURL,
+        name: ?[]const u8,
+        request: ?usize,
+        c_header: ?*polyfill_struct_curl_header() = null,
+
+        pub fn next(self: *HeaderIterator) ?Header {
+            if (comptime !has_parse_header_support()) return null;
```

Comment:
The compiler will still evaluate the content of `next()` which uses the possibly missing `curl_header` struct.

This check makes the compiler happy when using an old version of CURL.

Reply:
We should throw error in this case, not null.

Reply:
It's already throwing ;)

The first version did not throw because I don't think it's worth introducing errors in `next()`.

- As you mentioned, `iterateHeaders()` already fails in this case. This class is not intended to be created manually.
- `curl_easy_nextheader()` does not provide anyway to distinguish between error and end-of-headers anyway.

But now `curl_easy_header()` is used, it throws properly.